### ISSUE TITLE
Twisted overview graph shows Xish

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11524]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#1712|
|Type|defect|
|Created|2006-05-12 14:00:44Z|
|Branch|https://github.com/twisted/twisted/tree/overview-graph-1712|

The graphical overview at http://twistedmatrix.com/projects/core/documentation/howto/overview.html still shows Xish as a separate component. However, it has been folder into Words a while back.

Attachments:

* [graphical-overview.svg](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/aab/aab7f20827e516402f2387ba6f4f65adb26b54bc/e98f19bf4b02fc6a7584b08638a83e47f9e136e4.svg) (21322 bytes) - added by exarkun on 2011-01-22 21:51:05Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1712 1712
type__defect defect
reporter__ralphm ralphm
priority__normal normal
milestone__ 
branch__branches_overview_graph_1712 branches/overview-graph-1712
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1147442444000000 1147442444000000
changetime__1326807454000000 1326807454000000
version__ 
owner__ 
cc__exarkun cc__thijs cc__radix cc__binjured
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @teknico|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

It also shows Twisted.Flow, Twisted.Protocols as containing all the protocol implementations, half the executables in the wrong place, and implies "String ports" is a fundamental part of core twisted.  Basically half this picture is wrong.  The whole thing needs a going over.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm commented|
|-|-|

Would 2.3 be a good target for this?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_2):
> Basically half this picture is wrong.  The whole thing needs a going over.

Are the source files for those diagrams available?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

There's `doc/core/img/twisted-overview.dia`, that's probably it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [25338]) Branching to 'overview-graph-1712'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Removed xish and flow from the diagram in r25339, I'll need a better description/suggestion to restructure the diagram. Putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_9"></a>|@ralphm commented|
|-|-|

Your change indeed fixes the problem described in this ticket. Is your question whether this diagram should have more changes? I think so. Should it go into this ticket? I'd agree on widening the scope of this ticket.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @radix|
|-|-|

Yes.  The diagram still needs a lot of work.  Even with xish in the right place now, it doesn't really convey useful information to someone trying to learn about Twisted.  It's just a bunch of boxes and words and arrows.

We could remove it entirely or someone who's familiar with the various abstractions in Twisted could try fixing it (perhaps by starting over from scratch) to be useful.

A couple other specific problems with the current form:

  * it includes mktap, which we've stopped documenting anywhere else
  * it talks about bin/lore instead of bin/lore/lore

There are doubtless other issues, but those are the two which I noticed first.  Even if these are fixed, the diagram is still not useful (but the problems shouldn't be replicated in any future version).

As a roughly orthogonal issue, dia doesn't seem like the best format we could use for this kind of thing.  I'd suggest svg for a future version (but whatever).

Speculatively reassigning this to someone who might fall into the _familiar with the various abstractions in Twisted_ group.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

Assigning to radix for review then.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_12"></a>|@radix removed owner|
|-|-|

I'm not sure what I'm supposed to be reviewing here. It looks like this ticket is in need of implementation before I can review it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Let's include exarkun's suggestions and then hopefully merge:

 - it includes mktap, which we've stopped documenting anywhere else
 - it talks about bin/lore instead of bin/lore/lore

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

Other things that are wrong with the diagram:

  1. `twisted.protocols` no longer contains any of the protocols it talks about (except telnet, which is deprecated).  Instead it should actually be listing at least some of the stuff (ftp and socks, anyway) from the "Other" box in the "Twisted.Servers" section.
  1. "Twisted.Servers"?  What does that mean?  Those look like more "Twisted Subprojects", except for "Other", which I think we should just drop.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

I suggest we redo it in .svg so it can be opened in other programs besides Dia, which doesn't allow me to edit the text in the trunk's file.

**Comment by twisted-trac** [comment]
|<img alt="binjured's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_16"></a>|binjured removed owner<br>binjured set status to `new`|
|-|-|

If redoing this is underway or even the preferred method, might a sufficient resolution for this ticket be to remove the existing diagram from the documentation since the prevailing opinion seems to be that it isn't accurate or useful enough to do anything but confuse viewers? Then a redo could be undertaken as a different ticket with a preliminary number of corrections.

I don't know if flagging this for review is the right thing here, but the ticket seems to be at an impasse and the "Graphical Overview" is still the second link under /core/howto/ so the "path to resolution" should be decided.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun commented|
|-|-|

I started working on an SVG version of the overview (not really based on the existing one, which I don't think makes any interesting or informative organizational decisions).  Attached is the results of my efforts up to the point where I realized there's no way I'd actually finish it.

Removing the review keyword because there's still nothing to review here.  Someone needs to _make some graphics_, then there'll be something to review.

I don't think deleting the overview just so we can close a ticket is the right thing to do, though.  On the other hand, if someone wants to argue that a graphical overview isn't a useful part of the documentation, feel free (to argue that).

**Comment by twisted-trac** [comment]
|<img alt="binjured's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_19"></a>|binjured commented|
|-|-|

Replying to [exarkun](#note_18):
> I don't think deleting the overview just so we can close a ticket is the right thing to do, though.  On the other hand, if someone wants to argue that a graphical overview isn't a useful part of the documentation, feel free (to argue that). 

I didn't mean to argue that a graphical overview doesn't have the _potential_ to be a useful part of the documentation, merely that prevailing wisdom suggests the _current_ one is not. If something is uninformative at best and misleading at worst, it should probably just be removed until it's in a position not to do damage. Whether or not anybody else shares my opinion is another matter!

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_21"></a>|@thijstriemstra commented|
|-|-|

[#5432](https://github.com/twisted/twisted/issues/5432) was marked as a duplicate.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_22"></a>|@itamarst set status to `closed`|
|-|-|

(In [33453]) Merge howto-index-5427-2.

Author: itamar
Review: jesstess, exarkun, thijs
Fixes: [#5427](https://github.com/twisted/twisted/issues/5427), [#11524](https://github.com/twisted/twisted/issues/11524)

The core howto index has had many crufty howtos removed, and has been reorganized. Also, the generated PDF should now contain all howtos (it was missing some).
